### ISSUE TITLE
Creating Release with Specific Source Version or Build Number

### ISSUE THREAD

**Issue by benwalpole** [dmitryserbin/azdev-release-orchestrator#67]
Hi,

I've managed to get the extension to trigger a release against a specific branch but I want to be able to specify a specific build of that branch. I've tried using the Build Number of the build I want and the Source Version (i.e. Git commit hash) but I always get the message that no builds matching the filter can be found.

Is it possible to do this? I could just be using the wrong YAML parameters.

**Comment by dmitryserbin** [comment]
Hi, you may need to specify both parameters, then both filter will apply:

artifactVersion: YOUR_BUIL_NUMBER (i.e. 20220302.1)
artifactBranch: refs/heads/your_branch

I'm also pretty sure if you just specify build number without specifying branch filter - it will work just fine too as the build number is unique.

My guess it didn't work because use specify commit id instead of the build number.

Hope this helps!

**Comment by benwalpole** [comment]
Unfortunately I've tried adding both and just setting the Artifact Version to the Build Number but I get the same error in both cases.

Are you able to share any info on how the AzureDevOps API is being queried to implement the filtering?

**Comment by dmitryserbin** [comment]
Could you please run it with [debug mode on](https://github.com/dmitryserbin/azdev-release-orchestrator/blob/master/Docs/OrchestratorV2.md#troubleshooting) and share the logs?

**Comment by benwalpole** [comment]
Yeah sure, I don't fully understand the documentation for debug mode, do i just need to set System.Debug=true on the pipeline?

**Comment by dmitryserbin** [comment]
Either add `System.Debug=true` or `Debug=release-orchestrator:*` variable

**Comment by radulaurentiu02** [comment]
hello, are there any updates on this topic? I think I am facing the same issue. Thanks!
